src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -197,16 +201,149 @@ class SubcategorySelectionPools<SubgraphPool> {
   constructor(
     public pools: SubgraphPool[],
     public readonly poolsNeeded: number
-  ) {}
+  ) {
+  }
 
   public hasEnoughPools(): boolean {
     return this.pools.length >= this.poolsNeeded;
   }
 }
 
+export type CrossedLiquidityCandidatePools = {
+  v2Pools: V2SubgraphPool[];
+  v3Pools: V3SubgraphPool[];
+}
+
+export async function getMixedCrossLiquidityCandidatePools({
+  tokenIn,
+  tokenOut,
+  routingConfig,
+  v2SubgraphProvider,
+  v3SubgraphProvider,
+  v2Candidates,
+  v3Candidates,
+}: MixedCrossLiquidityCandidatePoolsParams): Promise<CrossedLiquidityCandidatePools> {
+  const blockNumber = routingConfig.blockNumber;
+
+  const v2Pools = (await v2SubgraphProvider.getPools(tokenIn, tokenOut, {
```

Comment:
you should be able to pull these from somewhere instead of fetching them all again right?

Reply:
In practice this doesn't matter. RoutingApi caches it in memory: https://github.com/Uniswap/routing-api/blob/0bc07607237678b1c2c18e752e05aa9f624bb4d2/lib/handlers/router-entities/aws-subgraph-provider.ts#L25-L48

I would have to do a larger refactor to allow it to be pulled only once, so I just went for the quickest solution here